tools/build-tools/src/test/typeValidator/classValidator.spec.ts

```diff
@@ -0,0 +1,131 @@
+            typeData = { kind: "unknown", name: "typeName", node: declarations[0] } as any as DecompositionTypeData;
+            tryDecomposeTypeData(project.getTypeChecker(), typeData);
+            break;
+        }
+        return typeData!;
+    }
+
+    // scenario: added new default param (method signature changed but won't break existing code)
+    // expected result: minor breaking change
+    it("new default value added", () => {
+        const classOld =
+        `
+        export class asdf {
+            public qewr() { return "afg"; }
+        }
+        `;
+        const oldSourceFile = project.createSourceFile(`${pkgDir}/src/classOld.ts`, classOld);
+        const oldTypeData = getTypeDataForSource(oldSourceFile);
+        // cont oldClassData = decomposeClassDeclaration(project.getTypeChecker(), oldSourceFile);
+
+        const classNew =
+        `
+        export class asdf {
+            public qewr(param1 : string = "afg") { return "afg"; }
+        }
+        `;
+        const newSourceFile = project.createSourceFile(`${pkgDir}/src/classNew.ts`, classNew);
+        const newTypeData = getTypeDataForSource(newSourceFile);
+
+        let increment = checkMajorIncrement(project, pkgDir, oldTypeData, newTypeData);
+        assert(increment === BreakingIncrement.none);
+
+        increment = checkMinorIncrement(project, pkgDir, oldTypeData, newTypeData);
+        assert(increment == BreakingIncrement.minor);
+
+    }).timeout(15000);
+
+
+    // scenario: new version changes the param to a different type.
+    // expected result: A major breaking change
+    it("default value type changed", () => {
+        const classOld =
+        `
+        export class asdf {
+            public qewr(param1 : string = "afg") { return "afg"; }
+        }
+        `;
+        const oldSourceFile = project.createSourceFile(`${pkgDir}/src/classOld.ts`, classOld);
+        const oldTypeData = getTypeDataForSource(oldSourceFile);
+        // cont oldClassData = decomposeClassDeclaration(project.getTypeChecker(), oldSourceFile);
+
+        const classNew =
+        `
+        export class asdf {
+            public qewr(param1 : boolean = false) { return "afg"; }
+        }
+        `;
+        const newSourceFile = project.createSourceFile(`${pkgDir}/src/classNew.ts`, classNew);
+        const newTypeData = getTypeDataForSource(newSourceFile);
+
+        let increment = checkMajorIncrement(project, pkgDir, oldTypeData, newTypeData);
+        assert(increment === BreakingIncrement.major);
+
+        increment = checkMinorIncrement(project, pkgDir, oldTypeData, newTypeData);
+        assert(increment == BreakingIncrement.minor);
+
+    }).timeout(15000);
+
+
+    // scenario: new version removes the default value from the param, changing
+    // the method signature. (optional param now required)
+    // expected result: A minor breaking change
+    it("default value removed", () => {
```

Comment:
this should be considered major breaking... it looks like ts is a bit more lenient with function param types compared to if you compared them directly, so we'll need to figure out something here so that's the case

Reply:
let's update this test to add a note on the behavior and expect the correct result (major), then either leave it failing or update it to not run (either with the mocha hook or by commenting it out) for now

Reply:
updated to major breaking and it is currently failing.

Reply:
update the expectation in the comment as well
also thinking about it some more can you use the mocha hook to not run this test instead of leaving it failing?  since it will be less disruptive if anyone does anything else here and the issue will be documented elsewhere anyway

Reply:
yeah make sense. I added a `.skip()` which will make the test case show up as `pending`